ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -260,19 +279,55 @@ private LogicalSchema rebuild(
       builder.add(Column.of(c.name(), c.type(), VALUE, valueIndex++));
     }
 
-    if (withPseudoAndKeyColsInValue) {
+    if (pseudoColumnVersion >= ROWTIME_PSEUDOCOLUMN_VERSION) {
       builder.add(Column.of(ROWTIME_NAME, ROWTIME_TYPE, VALUE, valueIndex++));
+    }
+
+    if (pseudoColumnVersion >= ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION) {
+      builder.add(Column.of(ROWOFFSET_NAME, ROWOFFSET_TYPE, VALUE, valueIndex++));
```

Comment:
This code adds ROWOFFSET then ROWPARTITION, while the code in SystemColumns.java returns pseudocolumn names with ROWPARTITION first instead. This is almost certainly going to lead to wacky bugs.

Reply:
Ok, I'm standardizing (including naming) to always make this offset before partition.